sdk/database/newdbplugin/middleware.go

```diff
@@ -0,0 +1,273 @@
+	defer func(now time.Time) {
+		metrics.MeasureSince([]string{"database", "DeleteUser"}, now)
+		metrics.MeasureSince([]string{"database", mw.typeStr, "DeleteUser"}, now)
+
+		if err != nil {
+			metrics.IncrCounter([]string{"database", "DeleteUser", "error"}, 1)
+			metrics.IncrCounter([]string{"database", mw.typeStr, "DeleteUser", "error"}, 1)
+		}
+	}(time.Now())
+
+	metrics.IncrCounter([]string{"database", "DeleteUser"}, 1)
+	metrics.IncrCounter([]string{"database", mw.typeStr, "DeleteUser"}, 1)
+	return mw.next.DeleteUser(ctx, req)
+}
+
+func (mw databaseMetricsMiddleware) Type() (string, error) {
+	return mw.next.Type()
+}
+
+func (mw databaseMetricsMiddleware) Close() (err error) {
+	defer func(now time.Time) {
+		metrics.MeasureSince([]string{"database", "Close"}, now)
+		metrics.MeasureSince([]string{"database", mw.typeStr, "Close"}, now)
+
+		if err != nil {
+			metrics.IncrCounter([]string{"database", "Close", "error"}, 1)
+			metrics.IncrCounter([]string{"database", mw.typeStr, "Close", "error"}, 1)
+		}
+	}(time.Now())
+
+	metrics.IncrCounter([]string{"database", "Close"}, 1)
+	metrics.IncrCounter([]string{"database", mw.typeStr, "Close"}, 1)
+	return mw.next.Close()
+}
+
+// ///////////////////////////////////////////////////
+// Error Sanitizer Middleware Domain
+// ///////////////////////////////////////////////////
+
+var _ Database = DatabaseErrorSanitizerMiddleware{}
+
+// DatabaseErrorSanitizerMiddleware wraps an implementation of Databases and
+// sanitizes returned error messages
+type DatabaseErrorSanitizerMiddleware struct {
+	next      Database
+	secretsFn secretsFn
+}
+
+type secretsFn func() map[string]string
+
+func NewDatabaseErrorSanitizerMiddleware(next Database, secrets secretsFn) DatabaseErrorSanitizerMiddleware {
+	return DatabaseErrorSanitizerMiddleware{
+		next:      next,
+		secretsFn: secrets,
+	}
+}
+
+func (mw DatabaseErrorSanitizerMiddleware) Initialize(ctx context.Context, req InitializeRequest) (resp InitializeResponse, err error) {
+	resp, err = mw.next.Initialize(ctx, req)
+	return resp, mw.sanitize(err)
+}
+
+func (mw DatabaseErrorSanitizerMiddleware) NewUser(ctx context.Context, req NewUserRequest) (resp NewUserResponse, err error) {
+	resp, err = mw.next.NewUser(ctx, req)
+	return resp, mw.sanitize(err)
+}
+
+func (mw DatabaseErrorSanitizerMiddleware) UpdateUser(ctx context.Context, req UpdateUserRequest) (UpdateUserResponse, error) {
+	resp, err := mw.next.UpdateUser(ctx, req)
+	return resp, mw.sanitize(err)
+}
+
+func (mw DatabaseErrorSanitizerMiddleware) DeleteUser(ctx context.Context, req DeleteUserRequest) (DeleteUserResponse, error) {
+	resp, err := mw.next.DeleteUser(ctx, req)
+	return resp, mw.sanitize(err)
+}
+
+func (mw DatabaseErrorSanitizerMiddleware) Type() (string, error) {
+	dbType, err := mw.next.Type()
+	return dbType, mw.sanitize(err)
+}
+
+func (mw DatabaseErrorSanitizerMiddleware) Close() (err error) {
+	return mw.sanitize(mw.next.Close())
+}
+
+// sanitize
```

Comment:
Missing full godoc description here?

Reply:
Originally copy-pasted from the old version of this middleware but I can flesh it out